cmake/Platform/Linux/CompilerSettings_linux.cmake

```diff
@@ -6,29 +6,30 @@
 #
 #
 
-if(NOT CMAKE_C_COMPILER AND NOT CMAKE_CXX_COMPILER AND NOT "$ENV{CC}" AND NOT "$ENV{CXX}")
-    set(path_search
-        /bin
-        /usr/bin
-        /usr/local/bin
-        /sbin
-        /usr/sbin
-        /usr/local/sbin
-    )
-    list(TRANSFORM path_search APPEND "/clang-[0-9]*")
-    file(GLOB clang_versions ${path_search})
-    if(clang_versions)
-        # Find and pick the highest installed version
-        list(SORT clang_versions COMPARE NATURAL)
-        list(GET clang_versions 0 clang_higher_version_path)
-        string(REGEX MATCH "clang-([0-9.]*)" clang_higher_version ${clang_higher_version_path})
-        if(CMAKE_MATCH_1)
-            set(CMAKE_C_COMPILER clang-${CMAKE_MATCH_1})
-            set(CMAKE_CXX_COMPILER clang++-${CMAKE_MATCH_1})
-        else()
-            message(FATAL_ERROR "Clang not found, please install clang")
-        endif()
-    else()
-        message(FATAL_ERROR "Clang not found, please install clang")
-    endif()
-endif()
+set(supported_clang_version_numbers 12 11 10 9 8 7 6.0)
+set(supported_clang_versions ${supported_clang_version_numbers})
+list(TRANSFORM supported_clang_versions PREPEND "clang-")
+set(supported_clangxx_versions ${supported_clang_version_numbers})
+list(TRANSFORM supported_clangxx_versions PREPEND "clang++-")
+
+find_program(
+    CMAKE_C_COMPILER
+    NAMES
+        $ENV{CC}
+        clang
+        ${supported_clang_versions}
+    REQUIRED
+)
+
+find_program(
+    CMAKE_CXX_COMPILER
+    NAMES
+        $ENV{CXX}
+        clang++
+        ${supported_clangxx_versions}
+    REQUIRED
+)
+
+unset(supported_clang_version_numbers)
+unset(supported_clang_versions)
+unset(supported_clangxx_versions)
```

Comment:
If you put the `find_program` calls in a function and the just invoked it, then the variables would have function scope and wouldn't need to be unset.